otherlibs/stdune/src/platform_stubs.c

```diff
@@ -0,0 +1,11 @@
+#include <caml/memory.h>
+#include <caml/mlvalues.h>
+
+CAMLprim value stdune_is_darwin(value v_unit) {
```

Comment:
We'll probably be able to reuse bits of this for the darwin detection in https://github.com/ocaml/dune/pull/7210 too

Reply:
Yes indeed. I'll update that PR after this.